types/index.d.ts

```diff
@@ -634,8 +639,8 @@ export default class Realm {
   define(
     name: string,
     attributes: Record<string, AttributeMeta>,
-    options: InitOptions,
-    descriptors: Record<string, Function>,
+    options?: InitOptions,
+    descriptors?: Record<string, Function>,
```

Comment:
descriptors aren't `Record<string, Function>`, the type of it is more like:

```ts
interface Descriptor {
  get() {},
  set() {},
  // not sure of the syntax above
}
Record<string, Descriptor>
```

It should be clear once you include some unit tests in test/types

Reply:
following syntax is possible as well, don't know how to express the type of it though:

```ts
Foo.init({}, {}, {
  get foo() {},
  set foo() {},
});
```

Reply:
> descriptors aren't `Record<string, Function>`, the type of it is more like:
> 
> ```ts
> interface Descriptor {
>   get() {},
>   set() {},
>   // not sure of the syntax above
> }
> Record<string, Descriptor>
> ```
> 
> It should be clear once you include some unit tests in test/types

This `descriptors` param seems to get passed in [getOwnPropertyDescriptors()](https://github.com/cyjake/leoric/blob/master/src/bone.js#L1553) call. It is a plain Object according to the [spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors).

Reply:
yep, the descriptors are more like `{ get foo() {} }` rather than full format property descriptors (which makes the parameter name misleading). 

Anyway, `Record<string, Function>` couldn't be right, take the `{ get foo() {} }` for example, the type of this object isn't `Record<string, Function>`, rather than something like `{ get [key: string](): any, set [key: string](): any }` (not sure if this works).

We can have the correct type definition of descriptor parameter once the test cases are ready.

Reply:
> yep, the descriptors are more like `{ get foo() {} }` rather than full format property descriptors (which makes the parameter name misleading).
> 
> Anyway, `Record<string, Function>` couldn't be right, take the `{ get foo() {} }` for example, the type of this object isn't `Record<string, Function>`, rather than something like `{ get [key: string](): any, set [key: string](): any }` (not sure if this works).
> 
> We can have the correct type definition of descriptor parameter once the test cases are ready.

Seems `getter/setter` are [not fully supported in TypeScript](https://github.com/microsoft/TypeScript/issues/3745#issuecomment-118924225) yet.

Reply:
yep, and looks like tsc is happy with `Record<string, Function>`. Let's focus on the rest of the changes, make sure these changes have test covered as well.